ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -631,3 +637,15 @@ def _make_union(union_class, expr, context):
         return union_class(
             table_exprs, expr, distincts=distincts, context=context
         )
+
+    @staticmethod
+    def _make_intersect(intersect_class, expr, context):
+        # flatten intersections so that we can codegen them all at once
+        table_exprs = list(flatten(expr))
+        return intersect_class(table_exprs, expr, context=context)
+
+    @staticmethod
+    def _make_difference(difference_class, expr, context):
+        # flatten differences so that we can codegen them all at once
+        table_exprs = list(flatten(expr))
+        return difference_class(table_exprs, expr, context=context)
```

Comment:
```suggestion
    @classmethod
    def _make_difference(cls, expr, context):
        # flatten differences so that we can codegen them all at once
        table_exprs = list(flatten(expr))
        return cls.difference_class(table_exprs, expr, context=context)
```

Similar to the above this should probably be a `@classmethod`.